docs/fundamentals/code-analysis/quality-rules/ca1861.md

```diff
@@ -26,60 +26,32 @@ dev_langs:
 
 ## Cause
 
-A constant array, of literal values, passed to a method via regular invocation or extension method invocation.
+A constant array of literal values, passed to a method via regular invocation or extension method invocation.
 
 ## Rule description
 
-Constant arrays passed as arguments aren't reused, which implies a performance overhead. Consider extracting them to `static readonly` fields to improve performance.
+Constant arrays passed as arguments are not reused when called repeatedly, which implies a new array is created each time. Consider extracting them to `static readonly` fields to improve performance if the passed array is not mutated within the called method.
 
-## How to fix violations
-
-Extract constant arrays as `static readonly` arrays.
-
-## When to suppress warnings
-
-Suppress a violation of this rule if you're not concerned about the performance impact of using constant arrays.
-
-## Suppress a warning
-
-If you just want to suppress a single violation, add preprocessor directives to your source file to disable and then re-enable the rule.
-
-```csharp
-#pragma warning disable CA1861
-// The code that's violating the rule is on this line.
-#pragma warning restore CA1861
-```
-
-To disable the rule for a file, folder, or project, set its severity to `none` in the [configuration file](../configuration-files.md).
-
-```ini
-[*.{cs,vb}]
-dotnet_diagnostic.CA1861.severity = none
-```
+  > [!NOTE]
+  > In case the called method mutates the passed array or if you are not sure if the method would mutate the array do not apply the suggestion as it could be a breaking change, better to suppress the warning instead.
```

Comment:
```suggestion
  > If the called method mutates the passed array or if you're not sure if the method would mutate the array, don't extract the array to a `static readonly` field. Doing so could be a breaking change. In this case, it's better to suppress the warning instead.
```

Reply:
Our style checker wants us to use contractions like "you're" instead of "you are".